ouroboros-consensus/src/Ouroboros/Consensus/Demo/Ledger/Mock.hs

```diff
@@ -27,38 +27,21 @@ instance HashAlgorithm h => DemoHeaderHash (Hash h a) where
   demoEncodeHeaderHash = Serialise.encode
   demoDecodeHeaderHash = Serialise.decode
 
-instance ( SupportedBlock (BlockProtocol (SimpleHeader SimpleMockCrypto ext))
-                          (SimpleHeader SimpleMockCrypto ext)
-         , Condense ext
-         , Typeable ext
-         , Serialise ext
-         ) => DemoHeader (SimpleHeader SimpleMockCrypto ext) where
-  demoEncodeHeader   = const Serialise.encode
-  demoDecodeHeader   = const Serialise.decode
-  demoBlockFetchSize = fromIntegral . simpleBlockSize . simpleHeaderStd
-
 instance ( ProtocolLedgerView (SimpleBlock SimpleMockCrypto ext)
-         , Typeable ext
-         , Condense ext
-         , Serialise ext
-         ) => DemoBlock (SimpleBlock SimpleMockCrypto ext) where
-  demoEncodeBlock = const Serialise.encode
-  demoDecodeBlock = const Serialise.decode
-  demoMockTx      = \_ -> SimpleGenTx
-
-instance ( SupportedBlock (BlockProtocol (SimpleHeader SimpleMockCrypto ext))
-                                         (SimpleHeader SimpleMockCrypto ext)
-         , ProtocolLedgerView (SimpleBlock  SimpleMockCrypto ext)
+           -- TODO: ProtocolLedgerView implies SupportedBlock already..?
+         , SupportedBlock (SimpleBlock SimpleMockCrypto ext)
```

Comment:
Huh, it compiles just fine if I remove this constraint.

Reply:
It didn't for me :-o